restart_plugin/mtcp_split_process.c

```diff
@@ -289,6 +289,15 @@ setLhMemRange(RestoreInfo *rinfo)
 #if !defined(MANA_USE_LH_FIXED_ADDRESS)
     lh_mem_range.start = (VA)area.addr - TWO_GB;
     lh_mem_range.end = (VA)area.addr - ONE_GB;
+    // High memory region occupies more than 1GB below stack region.
+    // Use minHighMemStart to set lower half memory range.
+    // Plugin_hook reserves region that ends 4GB below high memory region
+    // (See mtcp_plugin_hook). LH reserved 1GB size region that ends at 1GB
+    // below (high memory start - 4GB), so 5GB below high memory.
+    if (lh_mem_range.end > (VA) rinfo->minHighMemStart) {
+      lh_mem_range.end = rinfo->minHighMemStart - 5 * ONE_GB;
+      lh_mem_range.start = lh_mem_range.end - 1 * ONE_GB;
+    }
```

Comment:
@dahongli ,
    This is a suggestion.  But you've read the code and looked at the /proc/PID/maps more recently than I have.  So, I'm approving the PR now, and you can consider if you want to modify the PR.

QUESTION:  Should we add an MTCP_ASSERT after this, to make clear our _new_ assumption, and quickly catch any future bugs?

In particular, does `rinfo->minHighMemStart` have to be larger than some other address for a memory segment that is lower in the memory layout?  If so, let's specify this directly with an MTCP_ASSERT.

Reply:
@dahongli , Thanks for the assert.  This PR looks great.